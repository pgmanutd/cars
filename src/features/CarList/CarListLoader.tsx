import React from 'react';
import Grid from '@material-ui/core/Grid';
import Skeleton from '@material-ui/lab/Skeleton';

import CarCards from 'features/CarCards';

const CarListLoader: React.FC = () => (
  <Grid data-testid="CarListLoader" component="section" container spacing={2}>
    <Grid item md={12} xs={12}>
      <Skeleton height={40} width="30%" />
    </Grid>
    <Grid item md={8} xs={12}>
      <Skeleton height={40} width="50%" />
    </Grid>
    <Grid item md={4} xs={12}>
      <Skeleton height={80} />
    </Grid>
    <Grid item md={12} xs={12}>
      <CarCards cars={[]} isLoading />
    </Grid>
    <Grid item md={2} xs={12} />
    <Grid item md={8} xs={12}>
      <Skeleton height={40} />
    </Grid>
    <Grid item md={2} xs={12} />
  </Grid>
);

export default CarListLoader;
