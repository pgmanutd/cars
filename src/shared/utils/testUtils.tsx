import React from 'react';
import { Router } from 'react-router-dom';
import { createMemoryHistory, MemoryHistory } from 'history';
import { render } from '@testing-library/react';

export const renderWithRouter = (
  ui: React.ReactElement,
  {
    route = '/',
    history = createMemoryHistory({ initialEntries: [route] }),
  }: { route?: string; history?: MemoryHistory } = {},
) => ({
  ...render(<Router history={history}>{ui}</Router>),
  history,
});
