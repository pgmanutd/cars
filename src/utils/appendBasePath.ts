const appendBasePath = (basePath: string, path: string) =>
  `${basePath}?${path}`;

export default appendBasePath;
