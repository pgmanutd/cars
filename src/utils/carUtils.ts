import { Car } from 'features/CarInfo/types';

export const getCarName = ({ manufacturerName, modelName }: Car) =>
  `${manufacturerName} ${modelName}`;

export const getCarFeatures = (
  { stockNumber, mileage, fuelType, color }: Car,
  { stockNumberPrefix = '' }: { stockNumberPrefix?: string } = {
    stockNumberPrefix: '',
  },
) =>
  `${stockNumberPrefix}${stockNumber} - ${mileage?.number} ${mileage?.unit} - ${fuelType} - ${color}`;
