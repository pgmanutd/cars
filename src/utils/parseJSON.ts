const parseJSON = <T>(stringifiedJSON: string | null, fallbackValue: T) => {
  try {
    if (!stringifiedJSON) {
      return fallbackValue;
    }

    return JSON.parse(stringifiedJSON);
  } catch (e) {
    return fallbackValue;
  }
};

export default parseJSON;
