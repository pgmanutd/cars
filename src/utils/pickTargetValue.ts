import React from 'react';

const pickTargetValue = (event: React.ChangeEvent<{ value: unknown }>) =>
  event.target.value;

export default pickTargetValue;
