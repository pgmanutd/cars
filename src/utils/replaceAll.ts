export type ReplaceAllMappedObject = { [key: string]: string };

const replaceAll = (
  value: string,
  mappedObject: ReplaceAllMappedObject = {},
) => {
  const keys = Object.keys(mappedObject);

  if (!keys.length) {
    return value;
  }

  const regex = new RegExp(keys.join('|'), 'gi');

  return value.replace(regex, (matched) => {
    return mappedObject[matched];
  });
};

export default replaceAll;
