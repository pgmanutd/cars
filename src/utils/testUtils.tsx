import React from 'react';
import { Router, Route } from 'react-router-dom';
import { createMemoryHistory, MemoryHistory } from 'history';
import { HelmetProvider } from 'react-helmet-async';
import { render } from '@testing-library/react';

export const renderWithProviders = (
  ui: React.ReactElement,
  {
    routerConfig: {
      path = '/',
      route = '/',
      history = createMemoryHistory({ initialEntries: [route] }),
    },
  }: {
    routerConfig: { path?: string; route?: string; history?: MemoryHistory };
  } = { routerConfig: {} },
) => ({
  ...render(
    <HelmetProvider>
      <Router history={history}>
        <Route
          path={path}
          render={routeProps => React.cloneElement(ui, { ...routeProps })}
        />
      </Router>
    </HelmetProvider>,
  ),
  history,
});
