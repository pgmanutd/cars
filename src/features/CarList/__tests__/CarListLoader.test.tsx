import React from 'react';

import { renderWithTranslate } from 'utils/testUtils';

import CarListLoader from '../CarListLoader';

describe('<CarListLoader />', () => {
  const setup = () => {
    const { renderResult } = renderWithTranslate(<CarListLoader />);

    return {
      renderResult,
    };
  };

  it('should render the component and matches it against stored snapshot', () => {
    const { renderResult } = setup();

    expect(renderResult.asFragment()).toMatchSnapshot();
  });
});
