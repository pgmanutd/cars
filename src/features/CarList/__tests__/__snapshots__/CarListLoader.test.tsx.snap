// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CarListLoader /> should render the component and matches it against stored snapshot 1`] = `
<DocumentFragment>
  <section
    class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2"
    data-testid="CarListLoader"
  >
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-12"
    >
      <span
        class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse"
        style="width: 30%; height: 40px;"
      />
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-8"
    >
      <span
        class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse"
        style="width: 50%; height: 40px;"
      />
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-4"
    >
      <span
        class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse"
        style="height: 80px;"
      />
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-12"
    >
      <section
        class="MuiBox-root MuiBox-root-116"
        data-testid="CarCardsLoader"
      >
        <section
          class="MuiGrid-root makeStyles-carContainer-114 MuiGrid-container MuiGrid-spacing-xs-2"
          data-testid="CarCardLoader"
        >
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-2"
          >
            <span
              class="MuiSkeleton-root MuiSkeleton-rect MuiSkeleton-pulse"
              style="width: 100px; height: 85px;"
            />
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-5"
          >
            <span
              class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse"
              style="width: 80%; height: 32px;"
            />
            <span
              class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse"
              style="width: 80%; height: 20px;"
            />
            <span
              class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse"
              style="width: 30%; height: 16px;"
            />
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-5"
          >
            <span
              class="MuiSkeleton-root MuiSkeleton-rect MuiSkeleton-pulse"
              style="height: 150px;"
            />
          </div>
        </section>
        <section
          class="MuiGrid-root makeStyles-carContainer-114 MuiGrid-container MuiGrid-spacing-xs-2"
          data-testid="CarCardLoader"
        >
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-2"
          >
            <span
              class="MuiSkeleton-root MuiSkeleton-rect MuiSkeleton-pulse"
              style="width: 100px; height: 85px;"
            />
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-5"
          >
            <span
              class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse"
              style="width: 80%; height: 32px;"
            />
            <span
              class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse"
              style="width: 80%; height: 20px;"
            />
            <span
              class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse"
              style="width: 30%; height: 16px;"
            />
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-5"
          >
            <span
              class="MuiSkeleton-root MuiSkeleton-rect MuiSkeleton-pulse"
              style="height: 150px;"
            />
          </div>
        </section>
        <section
          class="MuiGrid-root makeStyles-carContainer-114 MuiGrid-container MuiGrid-spacing-xs-2"
          data-testid="CarCardLoader"
        >
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-2"
          >
            <span
              class="MuiSkeleton-root MuiSkeleton-rect MuiSkeleton-pulse"
              style="width: 100px; height: 85px;"
            />
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-5"
          >
            <span
              class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse"
              style="width: 80%; height: 32px;"
            />
            <span
              class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse"
              style="width: 80%; height: 20px;"
            />
            <span
              class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse"
              style="width: 30%; height: 16px;"
            />
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-5"
          >
            <span
              class="MuiSkeleton-root MuiSkeleton-rect MuiSkeleton-pulse"
              style="height: 150px;"
            />
          </div>
        </section>
        <section
          class="MuiGrid-root makeStyles-carContainer-114 MuiGrid-container MuiGrid-spacing-xs-2"
          data-testid="CarCardLoader"
        >
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-2"
          >
            <span
              class="MuiSkeleton-root MuiSkeleton-rect MuiSkeleton-pulse"
              style="width: 100px; height: 85px;"
            />
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-5"
          >
            <span
              class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse"
              style="width: 80%; height: 32px;"
            />
            <span
              class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse"
              style="width: 80%; height: 20px;"
            />
            <span
              class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse"
              style="width: 30%; height: 16px;"
            />
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-5"
          >
            <span
              class="MuiSkeleton-root MuiSkeleton-rect MuiSkeleton-pulse"
              style="height: 150px;"
            />
          </div>
        </section>
        <section
          class="MuiGrid-root makeStyles-carContainer-114 MuiGrid-container MuiGrid-spacing-xs-2"
          data-testid="CarCardLoader"
        >
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-2"
          >
            <span
              class="MuiSkeleton-root MuiSkeleton-rect MuiSkeleton-pulse"
              style="width: 100px; height: 85px;"
            />
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-5"
          >
            <span
              class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse"
              style="width: 80%; height: 32px;"
            />
            <span
              class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse"
              style="width: 80%; height: 20px;"
            />
            <span
              class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse"
              style="width: 30%; height: 16px;"
            />
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-5"
          >
            <span
              class="MuiSkeleton-root MuiSkeleton-rect MuiSkeleton-pulse"
              style="height: 150px;"
            />
          </div>
        </section>
      </section>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-2"
    />
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-8"
    >
      <span
        class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse"
        style="height: 40px;"
      />
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-2"
    />
  </section>
</DocumentFragment>
`;
